README.md

```diff
@@ -86,182 +105,299 @@ const UserProfile = ({ userId }) => (
+And we can `useResourceState` like normal.
 
 ```jsx
-const UserProfile = ({ userId }) => (
-  <RemoteResourceBoundary
-    /* Optional: A React node that will show while any thrown promises are pending. `null` by default. */
-    fallback={<p>Loading...</p>}
-    /* Optional: A callback that is invoked when any thrown promise rejects */
-    onLoadError={error => {
-      logError(error);
-    }}
-    /* Required: A render prop that receives the error and a function to clear the error, which allows the children to re-render and attempt loading again */
-    renderError={(error, clearError) => (
-      <div>
-        <p>{error}</p>
-        <button onClick={clearError}>Try again</button>
-      </div>
-    )}
-  >
-    <UserInfo userId={userId} />
-    <Tweets userId={userId} />
-  </RemoteResourceBoundary>
+const Tweets = ({ authToken, userId }) => {
+  const [tweetsByUserId] = useResourceState(tweetsResource, [
+    authToken,
+    userId
+  ]);
+
+  return tweetsByUserId[userId].map(tweet => (
+    <Tweet key={tweet.id} tweet={tweet} />
+  ));
+};
+```
+
+### `storeBy`
+
+A utility function that takes a hashing function and a load function. The state of the resource will now be an object literal where the keys are created by the hashing function. If a value does not exist at a given key then the load function will be called and the resolved value will be stored in the object state at the created key. This allows the data for a given key to only be fetched once.
+
+This is just an opinionated version of `withLens`.
+
+```javascript
+import { createRemoteResource, storeBy } from "react-remote-resource";
+
+const tweetsResource = createRemoteResource(
+  storeBy(
+    (authToken, userId) => userId,
+    (authToken, userId) =>
+      fetch(`/api/users/${userId}/tweets?auth=${authToken}`).then(res =>
+        res.json()
+      )
+  )
+);
+```
+
+### `withTimeout`
+
+A utility function that takes a timeout duration in milliseconds and a loader function. This allows you to prevent fetching for a certain amount of time. **Note:** The timeout will be bypassed if `refresh` is called.
+
+```javascript
+import { createRemoteResource, withTimeout } from "react-remote-resource";
+
+const tweetsResource = createRemoteResource(
+  withTimeout(10000, (currentState = {}) => (authToken, userId) =>
+    fetch(`/api/users/${userId}/tweets?auth=${authToken}`)
+      .then(res => res.json())
+      .then(tweets => ({ ...currentState, [userId]: tweets }))
+  )
 );
 ```
 
+### `useAutoSave`
+
+A React hook that takes a value, a save function, and an optional delay in milliseconds (defaults to 1000). When the value changes the new value will be saved if the value remains the same for the delay time.
+
+This is useful for optimistic UIs where the state of the resource is the source of truth and we are confident that the save will succeed.
```

Comment:
The `useAutoSave` function sounds exciting. Could the confidence of success become problematic for failures? An end user could handle them using a `try catch` or if a fetch request was made could use `.catch`.